client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningGetResultsIT.java

```diff
@@ -194,7 +206,73 @@ public void testGetBuckets() throws IOException {
         }
     }
 
-    private static class BucketStats {
+    public void testGetRecords() throws IOException {
+        MachineLearningClient machineLearningClient = highLevelClient().machineLearning();
+
+        {
+            GetRecordsRequest request = new GetRecordsRequest(JOB_ID);
+
+            GetRecordsResponse response = execute(request, machineLearningClient::getRecords, machineLearningClient::getRecordsAsync);
+
+            assertThat(response.count(), greaterThan(0L));
+            assertThat(response.count(), equalTo(recordStats.totalCount()));
+        }
+        {
+            GetRecordsRequest request = new GetRecordsRequest(JOB_ID);
+            request.setRecordScore(50.0);
+
+            GetRecordsResponse response = execute(request, machineLearningClient::getRecords, machineLearningClient::getRecordsAsync);
+
+            long majorAndCriticalCount = recordStats.majorCount + recordStats.criticalCount;
+            assertThat(response.count(), equalTo(majorAndCriticalCount));
+            assertThat(response.records().size(), equalTo((int) Math.min(100, majorAndCriticalCount)));
+            assertThat(response.records().stream().anyMatch(b -> b.getRecordScore() < 50.0), is(false));
```

Comment:
So, this could be simplified to `assertFalse` 

Or could be something like the following (which admittedly, is probably less simple)
```
import static org.hamcrest.CoreMatchers.everyItem;
import static org.hamcrest.Matchers.greaterThanOrEqualTo;
import static org.hamcrest.beans.HasPropertyWithValue.hasProperty;
...
assertThat(response.records(), everyItem(hasProperty("recordScore", greaterThanOrEqualTo(50))));
```

Man, that is frustrating that hamcrest does not support just passing a lambda as a type of matcher :(

Reply:
I try to avoid `assertFalse` as on failure the error message is not very helpful (it simply says `assertion failed` instead of giving some feedback on the type of assertion). I would prefer leaving as is.

Reply:
Fine with me, just a thought :)